controller/src/controller/models/domain/rsptag.py

```diff
@@ -402,10 +511,13 @@ def _compare(self, other: object) -> int:
             0 if equal, -1 if self is less than other, 1 if self is greater
             than other, `NotImplemented` if they're not comparable.
         """
-        if not isinstance(other, RSPImageTag):
-            return NotImplemented
-        if self.image_type != other.image_type:
+        try:
+            self._check_comparable_types(other)
+        except NotImplementedError:
             return NotImplemented
```

Comment:
A helper that returns a boolean seems simpler:
```suggestion
        if not self._is_comparable_type(other):
            return NotImplemented
```